app/src/main/java/com/example/grocerieswizard/ui/home/HomeFragment.java

```diff
@@ -109,129 +97,66 @@ public void onSwiped(RecyclerView.ViewHolder viewHolder, int direction) {
-                // User confirmed deletion, remove the recipe and update the RecyclerView
-                adapter.removeRecipe(recipeUi);
-                Toast.makeText(requireContext(), "Recipe deleted: " + recipeUi.getRecipeName(), Toast.LENGTH_SHORT).show();
-                dialog.dismiss();
-            });
-            builder.setNegativeButton("Cancel", (dialog, which) -> {
-                recipeUi.setSwiped(false);
-                adapter.itemChanged(position);
-                dialog.dismiss();
-            });
-            // Create and show the dialog
-            AlertDialog alertDialog = builder.create();
-            alertDialog.show();
-        }
+    public void onItemDelete(RecipeUi recipe) {
+        presenter.deleteRecipe(recipe);
     }
 
-    // Handle item edit and start AddRecipe activity for editing
     @Override
-    public void onItemEdit(int position) {
-        RecipeUi recipeUi = adapter.getItemAtPosition(position);
-        FragmentTransaction transaction = requireActivity().getSupportFragmentManager().beginTransaction();
-        AddRecipeFragment addRecipeFragment = new AddRecipeFragment();
-        Bundle bundle = new Bundle();
-        bundle.putParcelable("recipeModel", recipeUi);
-        bundle.putInt("position", position);
-        addRecipeFragment.setArguments(bundle);
-        transaction.replace(R.id.frameLayout, addRecipeFragment);
-        transaction.addToBackStack(null);
-        transaction.commit();
+    public void onItemEdit(RecipeUi recipe) {
+        showEditRecipe(recipe);
     }
 
-
     @Override
     public Boolean isRecipeSelected(int id) {
-        return recipeRepository.isRecipeInCart(id);
+        return presenter.isRecipeSelected(id);
     }
 
     @Override
     public int updateRecipe(RecipeUi oldRecipeUi) {
-        return recipeRepository.updateRecipe(oldRecipeUi.getId(), uiMapper.toRecipe(oldRecipeUi));
+        return presenter.updateRecipe(oldRecipeUi);
     }
 
     @Override
     public void insertRecipe(RecipeUi recipeUi) {
-        recipeRepository.insertRecipe(uiMapper.toRecipe(recipeUi));
+        presenter.insertRecipe(recipeUi);
     }
 
-    @Override
-    public void deleteRecipe(int id) {
-        Log.d(TAG, "deleteRecipe: id: " + id);
-        Log.d(TAG, "deleteRecipe: recipes before: " + recipeRepository.getAllRecipes().size());
-        recipeRepository.deleteRecipe(id);
-        Log.d(TAG, "deleteRecipe: recipes after: " + recipeRepository.getAllRecipes().size());
-    }
+
 
     @Override
     public void deleteSelectedRecipe(int recipeId) {
-        recipeRepository.deleteCartItem(recipeId);
+        presenter.deleteSelectedRecipe(recipeId);
     }
 
     @Override
     public void insertSelectedRecipe(int recipeId) {
-        CartItem cartItem = new CartItem(recipeId);
-        recipeRepository.insertCartItem(cartItem);
+        presenter.insertSelectedRecipe(recipeId);
     }
 
     @Override
     public boolean isRecipeFavorite(int id) {
```

Comment:
After you are done with the presenters we should fix this. You should add isFavorite and isSelected in RecipeUi model, so we don't have to call this method.